gcovr/gcov_parser.py

```diff
@@ -954,86 +967,155 @@ def line_after_start(self, lineno: int, start: str, start_lineno: str) -> None:
         )
 
 
+def _process_exclusion_marker(
+    code: str,
+    lineno: int,
+    exlcude_prefix: str,
+    exclude_word: str,
+    exclude_postfix: str,
+    exclude_by_pattern_regex: Pattern[str],
+    warnings: _ExclusionRangeWarnings,
+    exclude_ranges: List[Tuple[int, int]],
+    exclusion_stack: List[Tuple[int, int]],
+) -> Tuple[List[Tuple[int, int]], List[Tuple[int, int]]]:
+    """
+    Process the exclusion marker.
+
+    Header is a marker name like LCOV or GCOVR.
+
+    START flags are added to the exlusion stack
+    STOP flags remove a marker from the exclusion stack
+    """
+
+    if _EXCLUDE_FLAG in code:
+
+        # marker exclusion
+        excl_pattern = re.compile(
+            "(" + exlcude_prefix + ")" + _EXCLUDE_FLAG + exclude_word + exclude_postfix
+        )
+        for header, flag in excl_pattern.findall(code):
```

Comment:
I think it's better to keep this outside of the function to reduce the arguments.

Reply:
Fixed, I kept only what's in the for loop inside the function.